connect/tls.go

```diff
@@ -192,6 +191,9 @@ func verifyServerCertMatchesURI(certs []*x509.Certificate,
 	// domains.
 	expectURI := expected.URI()
 	expectURI.Host = gotURI.Host
+	// nolint: staticcheck // strings.EqualFold is the recommended way of doing
+	// this comparison, however it is more permissive, so keep the old comparison
+	// to preserve backwards compat.
 	if strings.ToLower(gotURI.String()) == strings.ToLower(expectURI.String()) {
```

Comment:
It would be nice to use `strings.EqualFold` here. More detail in https://staticcheck.io/docs/checks#SA6005.

Anyone know if comparing a few additional non-ascii characters as case-insensitive-equal would be a problem here? 

cc @banks

Reply:
:grimacing: IIRC it shouldn't be necessary because the comparison is of URI values which are already scoped to a restrictive character set where it doesn't matter.

Reply:
It sounds like the reason to use `EqualFold` is more about performance than allowing additional characters to compare. I'm not sure how important performance in this call.

So if EqualFold is equivalent because of the reduced character set, it sounds like it should be safe to use.

Reply:
Seems fine to swap to me.

Reply:
I doubt this will be a measurable performance gain as it happens once per TLS handshake and so probably the allocation/comparison cost of the old method is way less than the crypto work being done to verify the cert chain. but I'm also fine with standardising on EqualFold unless there is a good reason not to and I don't know of one here.

The additional permissiveness of supporting all unicode folding vs just ascii comparison if it made any difference seems to be correct to me but if I read x509 specs right I think the URI in a URI SAN must be an rfc3986 URI which can't include non-ascii chars anyway without percent-encoding.